src/App.js

```diff
@@ -1,17 +1,39 @@
+import { useState } from 'react'
 import styled from 'styled-components/macro'
 import LocationPage from './pages/LocationPage'
+import DetailsPage from './pages/DetailsPage'
 
 export default function App() {
+  const [activePage, setActivePage] = useState('LocationPage')
+  const [detailPage, setDetailPage] = useState(null)
+
   return (
     <AppGrid>
-      <LocationPage />
+      {activePage === 'LocationPage' && <LocationPage onDetail={showDetail} />}
+
+      {activePage === 'DetailsPage' && (
+        <DetailsPage
```

Comment:
At this point you just want to hand over the detailPage as detailPage and then de-structure it on the DetailsPage.
If you have questions, feel free to contact me.